src/Components/Aspire.Confluent.Kafka/MetricsService.cs

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Text.Json;
+using Microsoft.Extensions.Hosting;
+using static Aspire.Confluent.Kafka.ConfluentKafkaMetrics;
+
+namespace Aspire.Confluent.Kafka;
+
+internal sealed class MetricsService(MetricsChannel channel, ConfluentKafkaMetrics metrics) : BackgroundService
+{
+    private readonly MetricsChannel _channel = channel;
+    private readonly Dictionary<string, Statistics> _state = new();
+    private readonly ConfluentKafkaMetrics _metrics = metrics;
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (await _channel.Reader.WaitToReadAsync(stoppingToken).ConfigureAwait(false))
+        {
+            while (_channel.Reader.TryRead(out var json))
+            {
+                Statistics? statistics;
+                try
+                {
+                     statistics = JsonSerializer.Deserialize(json, StatisticsJsonSerializerContext.Default.Statistics);
+                }
+                catch
+                {
+                    return;
+                }
+
+                if (statistics == null || statistics.Name == null)
+                {
+                    return;
```

Comment:
Do we really want to stop the MetricsService if one bad JSON document got in the channel?

Do we want to log this?

Reply:
done. I added a warning log and replaced the return by a continue statement